File: 217_Contains_Duplicate.js
var containsDuplicate = function(nums) {
    nums = nums.sort();
    let i = 1;
    let temp = nums[0];
    while(i<nums.length){
        if(temp !== nums[i]){
            temp = nums[i];
        }
        else return true;
        i++;
    }
    if(i === nums.length) return false;
};

File: 217_Contains_Duplicate.cpp
class Solution {
public:
    bool containsDuplicate(vector<int>& nums) {
        unordered_map<int,int> mp;
        mp[nums[0]] = 1;
        int i = 1;
        while(i < nums.size()){
            auto it = mp.find(nums[i]);
            if(it != mp.end()){
                return true;
            }
            else{
                mp[nums[i]] = 1;
            }
            i++;
        }
        return false;
    }
};
